packages/components/src/components/Grid/index.tsx

```diff
@@ -0,0 +1,48 @@
+import styled from 'styled-components';
+import css from '@styled-system/css';
+
+const fontSize = 1; // rem = 16px
+const lineHeight = fontSize * 1.5;
+
+export const Grid = styled.div<{ columnGap?: number; rowGap?: number }>(
+  ({ columnGap, rowGap }) =>
+    css({
+      display: 'grid',
+      gridTemplateColumns: 'repeat(12, 1fr)', // always 12 columns
+      gridColumnGap:
+        (typeof columnGap !== 'undefined' ? columnGap : lineHeight * 2) + 'rem',
+      gridRowGap: (typeof rowGap !== 'undefined' ? rowGap : lineHeight) + 'rem',
+    })
+);
+
+// todo: end and span cant be together
+// valid combinations are
+// start | start + end | start + span | span
+// span + end is also possible but not implemented here
+export const Column = styled.div<{
```

Comment:
```suggestion
type ColumnEnd = number | number[];
type ColumnSpan = number | number[];
type ColumnStart = number | Array<number | string>;
type ColumnProps =
  | { end: ColumnEnd; start: ColumnStart; }
  | { span: ColumnSpan; }
  | { span: ColumnSpan; start: ColumnStart; }
  | { start: ColumnStart; };
export const Column = styled.div<ColumnProps>(
```

Reply:
what does this improve?

Reply:
This is implementing the Props like you said in your comments 🙂

Reply:
Got it. Implementing it this way would give errors on the missing prop in each, right?

repro (see end prop): https://codesandbox.io/s/funny-pare-zswfv

Reply:
+ This is not a blocker, so I'm merging the PR but will continue to improve the types